File: exercises/small_problems_101/easy_1/how_many.rb
def count_occurrences(array)
  results = {}
  array.each do |element|
    if results.has_key?(element)
      results[element] += 1
    else
      results[element] = 1
    end
  end
  results.each_pair do |key, value|
    puts "#{key} => #{value}"
  end
end

vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck', 'motorcycle', 'motorcycle', 'car', 'truck']
count_occurrences(vehicles)

File: frontend/javascript/exercises/easy_4/how_many.js
function countOccurrences(words) {
  var counts = {}
  words.forEach(function (word) {
    counts[word] = counts[word] || 0;
    counts[word]++;
  });
  Object.keys(counts).forEach (function (word) {
    console.log(word + ' => ' + counts[word]);
  });
}

var vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck', 'motorcycle', 'motorcycle', 'car', 'truck'];
countOccurrences(vehicles);